utils/ort/src/main/kotlin/CopyrightStatementsProcessor.kt

```diff
@@ -31,7 +31,11 @@ private data class Parts(
     val years: Set<Int>,
     val owner: String,
     val originalStatements: List<String>
-)
+) {
```

Comment:
In my view the comparator belongs to `CopyrightStatementsProcesser` because it defines how `CopyrightStatementsProcesser` orders its results. So, moving it to companion object to `CopyrightStatementsProcesser` clarifies that ownership IMO better and also enables sharing. E.g. one can just change the ordering `CopyrightStatementsProcesser` without worrying about (custom) calling code.

What do you think?

Reply:
> In my view the comparator belongs to `CopyrightStatementsProcesser` because it defines how `CopyrightStatementsProcesser` orders its results.

I can see the point that there is some subjectivity to *how* `Parts` should be sorted, and that the current ordering might be seen as `CopyrightStatementsProcesser`-specific. However, in general I believe a data class knows best for itself how it should be sorted. At least it should have a default, expressed as `Comparable`. Otherwise such data classes could not be put into sorted sets or be used as keys in sorted maps, which is functionality I rely on in the later changes.

> It'll become harder to figure out which callers get affected when changing the ordering.

True, but I guess that's a rather theoretical argument as the ordering has been the same for years, and I don't foresee any changes to it.

> I'm approving this anyway, as you seem to be on a rush but would prefer if it was changed.

I'm not on a "rush", but I simply believe that discussing 4 days with > 30 comments over a small change that basically just moves existing code is enough. So, thanks for approving finally.